### PR TITLE
Flex provisioner example.

### PR DESCRIPTION
**Pull request by childsb**
Move the flex provisioner example from https://github.com/childsb/flex-provision

### CODE REVIEW

**Comment by wongma7** [comment]
it won't build with the shared vendor dir after we merge it because they reshuffled client-go and apimachinery repos. Need to rebase on latest external-storage master and fix the errors. I can fix it after we merge

File: flex/deploy/docker/flex-provision.sh [hunk 1]
```diff
@@ -0,0 +1,119 @@
+#!/bin/sh
+
```
Comment (wongma7) [inline flex/deploy/docker/flex-provision.sh]: script is complaining this bash file needs boilerplate

File: flex/pkg/volume/provision_test.go [hunk 2]
```diff
@@ -0,0 +1,639 @@
+/*
```
Comment (wongma7) [inline flex/pkg/volume/provision_test.go]: this file should be deleted I think it has a lot of nfs provisioner specific things still

**Comment by childsb** [comment]
I made the two updates you requested.  As far as the build.....

It works from me building clean--

```bash
mkdir -p /opt/go/src/github.com/kubernetes-incubator/
git clone git@github.com:childsb/external-storage.git
cd /opt/go/src/github.com/kubernetes-incubator/external-storage/flex
make
```

Where are you building that fails because of the shared library?

**Pull request merged**